lib/auth/bot.go

```diff
@@ -258,6 +327,30 @@ func (a *Server) updateBotInstance(
 			return trace.Wrap(err)
 		}
 
+		// TODO: consider making this the only path where bot instances are
```

Comment:
This would be nice - but perhaps let's wait until we're pulling out the experiment conditionals so we can simplify the whole thing at once.

Reply:
Agreed, I think (as we discussed offline) we'll have a nice opportunity to refactor/reorganize this giant function when the experiment flag is removed.